src/qmapshack/gis/rte/router/CRouterRoutino.cpp

```diff
@@ -269,12 +296,29 @@ void CRouterRoutino::freeDatabaseList()
 {
     for(int i = 0; i < comboDatabase->count(); i++)
     {
-        Routino_Database * data = (Routino_Database*)comboDatabase->itemData(i, Qt::UserRole).toULongLong();
+        QVariantMap map = comboDatabase->itemData(i,Qt::UserRole).toMap();
+        Routino_Database * data = (Routino_Database*)(map["db"].toULongLong());
         Routino_UnloadDatabase(data);
     }
     comboDatabase->clear();
 }
 
+void CRouterRoutino::loadProfiles(QString profilesPath)
+{
+    // load xml profiles if it is new before setting the profile!
+    if( currentProfilesPath != profilesPath)
+    {
+        currentProfilesPath = profilesPath;
+        int res = 0;
+        res = Routino_ParseXMLProfiles(profilesPath.toStdString().data());
+        if(res)
+        {
+            QMessageBox::critical(this, "Routino...", xlateRoutinoError(Routino_errno)+"\n '"+profilesPath+"'", QMessageBox::Abort);
+            abort();
```

Comment:
Abort execution and create a coredump? That's a quite drastic way to handle that kind of error.

Reply:
This was done because it leads to seg errors in the routing calculations.  These are not wanted.  The only way I can see to get around this is to test load the profiles  in the buildDatabaseList() call.  This is being implemented.  In the mean time leave this abort here but should never be called if the errors are caught earlier.  Suggestions welcomed.

Reply:
OK I now have loadProfiles called during the buildDatabaseList which catches any profile errors.  But what is the recommended way to smoothly exit the program without coredumps etc - use throw QString() also does a coredump??  What do you recommend for the button to close the window, I would have thought OK if I add extra text to indicate the  file needs fixing?  Please your suggestion.

Reply:
Never terminate the program the hard way. You will destroy the users unsaved data. No user will appreciate that. This is not a command line application you can abort on an error to fall back to command line. This is a very complex GUI application that should not be closed by no one else than the user by intention. Everything else is considered a bug. 

That in mind: Inform the user that something is wrong. Give as much information to help the user to fix it. After all we are in that condition because the user did something peculiar. It's also ok to add a QLabel to the GUI that tells in red fat letters that routing is not available due to errors. BRouter has such a label. Additionally you have to block test in the API of CRouterRoutino if routing is possible. If not simply return with no result. 

In other words: In the error case every thing should work as expected. Just the faulty function is not working.Ideally there is some feedback to inform the user.